### DISCUSSION TITLE
Search results aren't displayed unless I press up or down arrow key

### DISCUSSION THREAD

**Discussion by xfzv** [mrjones2014/legendary.nvim#341] [Q&A] [answered]
This is definitely an issue with my config because I cannot reproduce with the [minimal `init.lua`](https://github.com/mrjones2014/legendary.nvim/discussions/93) provided.

I'm using `stevearc/dressing.nvim` and `nvim-telescope/telescope.nvim` as `vim.ui.select()` handlers.

1. Run `:Legendary`
2. Start typing

### GIFs showcasing the two different behaviors

- With minimal `init.lua` :heavy_check_mark:  - all good (results start appearing as I type)
![minimal](https://user-images.githubusercontent.com/78810647/229528798-7523ae24-56a5-4317-b09e-68dd48b38ae8.gif)

- With my main config :x:  (pressing down arrow key at the very end so that search results are displayed)
![main](https://user-images.githubusercontent.com/78810647/229534794-f842f32c-560f-43b5-a797-3782e85b5cd0.gif)

---

<details>
  <summary>minimal init.lua:</summary>
   
  ```lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])
local package_root = "/tmp/nvim/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
vim.g.loaded_remote_plugins = ""
local function load_plugins()
	require("packer").startup({
		{
			"wbthomason/packer.nvim",
			{
				"mrjones2014/legendary.nvim",
				requires = {
					-- place your `vim.ui.select()` handler here
					-- for example:
					{
						"nvim-telescope/telescope.nvim",
						requires = "nvim-lua/plenary.nvim",
					},
					"stevearc/dressing.nvim",
				},
			},
			-- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
		},
		config = {
			package_root = package_root,
			compile_path = install_path .. "/plugin/packer_compiled.lua",
			display = { non_interactive = true },
		},
	})
end
_G.load_config = function()
	-- setup your `vim.ui.select()` handler here
	-- for example:
	require("telescope").setup()
	require("dressing").setup()
	require("legendary").setup()
	-- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
	print("Installing legendary.nvim and dependencies.")
	vim.fn.system({ "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path })
end
load_plugins()
require("packer").sync()
vim.cmd([[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]])
  ```
</details>

<details>
  <summary>relevant parts of my main config (I'm using lazy.nvim)</summary>
   
  ```lua
	{
		"mrjones2014/legendary.nvim",
		config = true,
	},
{
		"nvim-telescope/telescope.nvim",
		cmd = "Telescope",
		dependencies = {
			{
				"nvim-lua/plenary.nvim",
				lazy = true,
			},
			{
				"nvim-lua/popup.nvim",
				lazy = true,
			},
			"tsakirist/telescope-lazy.nvim",
			{
				"jvgrootveld/telescope-zoxide",
			},
		},
		config = function()
			require("telescope").setup({
				defaults = {
					find_command = {
						"rg",
						"--files",
						"--hidden",
						"--ignore",
						"--line-number",
						"--smart-case",
					},
					prompt_prefix = "🔍 ",
					selection_caret = " ",
					vimgrep_arguments = {
						"rg",
						"--color=auto",
						"--column",
						"--hidden",
						"--line-number",
						"--no-heading",
						"--smart-case",
						"--with-filename",
					},
				},
				require("telescope").load_extension("lazy"),
				extensions = {
					zoxide = {
						prompt_title = "Zoxide",
						mappings = {
							default = {
								after_action = function(selection)
									print("Directory changed to (" .. selection.z_score .. ") " .. selection.path)
								end,
							},
							["<CR>"] = {
								action = require("telescope._extensions.zoxide.utils").create_basic_command("cd"),
							},
						},
					},
				},
			})
		end,
	},
{
		"stevearc/dressing.nvim",
		lazy = true,
		init = function()
			vim.ui.select = function(...)
				require("lazy").load({ plugins = { "dressing.nvim" } })
				return vim.ui.select(...)
			end
			vim.ui.input = function(...)
				require("lazy").load({ plugins = { "dressing.nvim" } })
				return vim.ui.input(...)
			end
		end,
	},
  ```
</details>

Other information about my setup:

```console
❯ nvim --version
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3

❯ kitty --version
kitty 0.27.1 created by Kovid Goyal
```

Any idea what could be causing this?
Thanks!

**Comment by mrjones2014** [comment, marked as answer]
Taking a brief look, the only thing that stands out is that `require("telescope").load_extension("lazy")` is in the middle of the telescope config table? Shouldn't that be outside of the `require("telescope").setup()` table?

**Reply by xfzv** [reply]
Indeed! I removed that useless line, thanks.

It doesn't make any difference unfortunately, same issue.

**Reply by xfzv** [reply]
I've just noticed the same issue while using Telescope so it's not specific to `legendary.nvim`. Any insight is still appreciated, of course.

**Reply by mrjones2014** [reply]
I’d recommend removing telescope extensions and adding them back one by one to rule out a telescope extension causing it.

**Reply by xfzv** [reply]
I tried without any Telescope extension and without any additional parameters but the issue persists:

```lua
	{
		"nvim-telescope/telescope.nvim",
		config = true,
		dependencies = {
			"nvim-lua/plenary.nvim",
			"nvim-lua/popup.nvim",
		},
	}
```



 Without Telescope and while keeping `dressing.nvim`, the issue doesn't occur.

I'll have to do some further testing. Thanks for your help!

**Reply by xfzv** [reply]
I found the culprit, maybe someone will find this useful. I'm using [NvChad/nvim-colorizer.lua](https://github.com/NvChad/nvim-colorizer.lua):

```lua
	{
		"NvChad/nvim-colorizer.lua",
		event = { "BufReadPost", "BufNewFile" },
		config = function()
			require("colorizer").setup({
				user_default_options = {
                                        -- update color values even if buffer is not focused
                                        -- example use: cmp_menu, cmp_docs
					always_update = true,
				},
			})
		end,
	},
```

Switching `always_update` to `false` ([which is the default by the way](https://github.com/NvChad/nvim-colorizer.lua/blob/master/doc/colorizer.txt#L248-L249)) did the trick.

Thanks again for your time @mrjones2014.

**Reply by mrjones2014** [reply]
Cool, I recommend filing an issue in their repo.

**Reply by mrjones2014** [reply]
I just tried it out and I don't have the issue with that plugin and `always_update = true`
